src/dataset/infrastructure/repositories/DatasetJSDataverseRepository.ts

```diff
@@ -161,8 +179,11 @@ export class DatasetJSDataverseRepository implements DatasetRepository {
           datasetDetails.jsDataset.publicationDate !== undefined
         ) {
           // If the dataset is a draft, but has a publication date, then we need the version
-          // numbers of the latest published version to show in the "Publish" button
-          return this.getLatestPublishedVersionNumbers(datasetDetails)
+          // numbers of the latest published version and the datasetVersionDiff,
+          // for the PublishDatasetModal component.
+          return this.getLatestPublishedVersionNumbers(datasetDetails).then((updatedDetails) =>
+            this.getVersionDiffDetails(updatedDetails)
+          )
```

Comment:
What about if the dataset is in draft but it has a publication date? then you are not setting the datasetVersionDiff to the dataset info right?

```

.then((datasetDetails) => {
  if (
    datasetDetails.jsDataset.versionInfo.state === DatasetVersionState.DRAFT &&
    datasetDetails.jsDataset.publicationDate !== undefined
  ) {
    // If the dataset is a draft, but has a publication date, then we need the version
    // numbers of the latest published version and the datasetVersionDiff,
    // for the PublishDatasetModal component.
    return this.getLatestPublishedVersionNumbers(datasetDetails).then((updatedDetails) =>
      this.getVersionDiffDetails(updatedDetails)
    )
  } else {
    return datasetDetails
  }
})
```

Reply:
I'm not sure exactly your question, but we need to get the versionDiff only if the dataset was previously released, and the current (requested) version is a draft. In that case, we use the versionDiff to determine if a major version update is needed.  
If there is no publicationDate, that means that only one version exists, which is the draft version, so when publishing, the version is always set to 1.0.

Reply:
great, thanks! thats exactly what I wanted to know 👍🏼